src/utils/RNPxlSearch.cpp

```diff
@@ -985,18 +985,16 @@ class RNPxlSearch :
           // generate total loss spectrum for the fixed and variable modified peptide (without RNA)
           PeakSpectrum total_loss_spectrum;
 
-          // TODO for ETD: generate MS2 precursor peaks of the MS1 adduct (total RNA) carrying peptide for all z <= precursor charge  
+          // TODO for ETD: generate MS2 precursor peaks of the MS1 adduct (total RNA) carrying peptide for all z <= precursor charge
 
-          for (Size z = 1; z <= precursor_charge; ++z)
-          {
-            spectrum_generator.addPeaks(total_loss_spectrum, fixed_and_variable_modified_peptide, Residue::AIon, z);
-            spectrum_generator.addPeaks(total_loss_spectrum, fixed_and_variable_modified_peptide, Residue::BIon, z);
-            spectrum_generator.addPeaks(total_loss_spectrum, fixed_and_variable_modified_peptide, Residue::YIon, z);
-            // generate MS2 precursor peaks of the total loss peptide for all z <= precursor charge  
-            spectrum_generator.addPrecursorPeaks(total_loss_spectrum, fixed_and_variable_modified_peptide, z);
-          }
-
-          total_loss_spectrum.sortByPosition();
+          Param params;

```

Comment:
might be quite costly to set this in the loop. Maybe create two spectrum generators outside the loop?